### ISSUE TITLE
Build error when trying to build mmdblookup module: mmdblookup.c:245:6: error: unknown type name ‘msg_t’

### ISSUE THREAD

**Issue by Whissi** [rsyslog/rsyslog#1355]
```
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I../.. -I../../runtime -I../.. -I../../grammar -I/usr/include/libfastjson -W -Wall -Wformat-security -Wshadow -Wcast-align -Wpointer-arith -Wmissing-format-attribute -Werror=implicit-function-declaration -fno-strict-aliasing -Wall -Wextra -Wundef -Wnested-externs -Wwrite-strings -Wpointer-arith -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wno-unused-parameter -Wno-missing-field-initializers -Wdeclaration-after-statement -Wformat=2 -Wold-style-definition -Wcast-align -Wformat-nonliteral -Wformat-security -Wsign-compare -Wstrict-aliasing -Wshadow -Winline -Wpacked -Wmissing-format-attribute -Wmissing-noreturn -Winit-self -Wmissing-include-dirs -Wunused-but-set-variable -Warray-bounds -Wimplicit-function-declaration -Wreturn-type -Wswitch-enum -Wswitch-default -Wno-error=unused-parameter -Wno-error=missing-field-initializers -pthread -O2 -pipe -march=ivybridge -mtune=ivybridge -mno-xsaveopt -frecord-gcc-switches -ggdb -c mmdblookup.c  -fPIC -DPIC -o .libs/mmdblookup_la-mmdblookup.o
mmdblookup.c:213:6: warning: no previous prototype for ‘str_split’ [-Wmissing-prototypes]
 void str_split(char **membuf){
      ^
mmdblookup.c: In function ‘doAction’:
mmdblookup.c:245:6: error: unknown type name ‘msg_t’
      msg_t *pMsg;
      ^
mmdblookup.c:251:14: error: ‘msg_t’ undeclared (first use in this function)
      pMsg = (msg_t*) ppString[0];
              ^
mmdblookup.c:251:14: note: each undeclared identifier is reported only once for each function it appears in
mmdblookup.c:251:20: error: expected expression before ‘)’ token
      pMsg = (msg_t*) ppString[0];
                    ^
mmdblookup.c:258:45: warning: passing argument 1 of ‘msgGetJSONPropJSON’ from incompatible pointer type [-Wincompatible-pointer-types]
      rsRetVal localRet = msgGetJSONPropJSON(pMsg, &pProp, &keyjson);
                                             ^
In file included from ../../template.h:73:0,
                 from ../../runtime/msg.h:26,
                 from ../../runtime/batch.h:31,
                 from ../../runtime/queue.h:30,
                 from ../../action.h:29,
                 from ../../runtime/conf.h:23,
                 from mmdblookup.c:34:
../../runtime/msg.h:216:10: note: expected ‘smsg_t * {aka struct msg *}’ but argument is of type ‘int *’
 rsRetVal msgGetJSONPropJSON(smsg_t *pMsg, msgPropDescr_t *pProp, struct json_object **pjson);
          ^
mmdblookup.c:319:15: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   char *sep = "!";
               ^
mmdblookup.c:326:4: warning: ‘fjson_object_object_get’ is deprecated [-Wdeprecated-declarations]
    json_object *sub_obj = json_object_object_get(temp_json, path[j]);
    ^
In file included from /usr/include/libfastjson/json_util.h:13:0,
                 from /usr/include/libfastjson/json.h:20,
                 from ../../grammar/rainerscript.h:25,
                 from ../../runtime/modules.h:39,
                 from ../../runtime/obj-types.h:423,
                 from ../../runtime/debug.h:28,
                 from ../../runtime/rsyslog.h:623,
                 from mmdblookup.c:24:
/usr/include/libfastjson/json_object.h:349:57: note: declared here
 THIS_FUNCTION_IS_DEPRECATED(extern struct fjson_object* fjson_object_object_get(struct fjson_object* obj,
                                                         ^
/usr/include/libfastjson/json_object.h:16:43: note: in definition of macro ‘THIS_FUNCTION_IS_DEPRECATED’
 #define THIS_FUNCTION_IS_DEPRECATED(func) func __attribute__ ((deprecated))
                                           ^
mmdblookup.c:348:22: warning: passing argument 1 of ‘msgAddJSON’ from incompatible pointer type [-Wincompatible-pointer-types]
           msgAddJSON(pMsg, (uchar *)JSON_IPLOOKUP_NAME, json, 0, 0);
                      ^
In file included from ../../template.h:73:0,
                 from ../../runtime/msg.h:26,
                 from ../../runtime/batch.h:31,
                 from ../../runtime/queue.h:30,
                 from ../../action.h:29,
                 from ../../runtime/conf.h:23,
                 from mmdblookup.c:34:
../../runtime/msg.h:196:10: note: expected ‘smsg_t * {aka struct msg *}’ but argument is of type ‘int *’
 rsRetVal msgAddJSON(smsg_t *pM, uchar *name, struct json_object *json, int force_reset, int sharedReference);
          ^
In file included from mmdblookup.c:38:0:
../../runtime/module-template.h:281:10: warning: unused variable ‘ppString’ [-Wunused-variable]
  uchar **ppString = (uchar **) pMsgData; \
          ^
mmdblookup.c:244:1: note: in expansion of macro ‘BEGINdoAction’
 BEGINdoAction
 ^
make[2]: *** [Makefile:528: mmdblookup_la-mmdblookup.lo] Error 1
make[2]: Leaving directory '/var/tmp/portage/app-admin/rsyslog-8.24.0/work/rsyslog-8.24.0/contrib/mmdblookup'
make[1]: *** [Makefile:807: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/app-admin/rsyslog-8.24.0/work/rsyslog-8.24.0'
make: *** [Makefile:634: all] Error 2
```

I was trying to build against libfastjson-0.99.4.

**Comment by Whissi** [comment]
Wondering why this contribution was merged without test coverage. It is disabled in your Travis configuration so you haven't noticed...

**Comment by rgerhards** [comment]
Two things:

(1) The general policy for **contributed** modules is that we do not check anything (except a brief review) nor guarantee anything. We also do not require any test coverage, as that usually would mean the contribution cannot be merged. The contrib tree should really be considered broken except when it works (user needs to verify).

(2) in this special case, I even did some review, and it would make a lot of sense to at least build it during CI testing. I should have done this! Unfortunately, I didn't have the time when I merged. And I overlooked that the AIX patches changes some core data structure names :-(

Number (1) is just to clarify the expectations, number (2) tells that I f*cked up in this case and will fix in the next cycle ;-)

Thanks for adding the issue tracker!
